hearts/hearts.py

```diff
@@ -204,7 +204,7 @@ def upkeep(self, player, card):
 
     def play_card(self, player, card):
         if self.is_player_turn(player):
-            if len(self.tricks) == 0 or len(self.tricks[-1]) == 4:
+            if len(self.tricks) == 0 or self.tricks[-1].size == 4:
```

Comment:
So this is actually a bug that's currently in master. Cf. this line of Trick:

https://github.com/j2kun/lonely-hearts/blob/master/hearts/hearts.py#L228

The Trick only computes the size at the time the Trick is created, and so the `.size` attribute doesn't update when you do `trick.cards_played.append`.

There are two ways to fix this:

1. Add a helper function to `Trick` called `add` that adds a new card and updates and required state.
2. (Preferred) Define `__len__` as a method on `Trick` that returns `len(cards_played)`.

I'm going to leave this as is, but then open an issue with the bug and assign it to you.